.github/workflows/run-pytest.yml

```diff
@@ -16,39 +16,67 @@ concurrency:
   cancel-in-progress: true
 jobs:
   test:
+    name: test (${{ matrix.os }}, ${{ matrix.python }}, ${{ matrix.dependencies }})
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
-        config: [ {python: '3.9', dependencies: 'newest'},
-                  {python: '3.10', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'newest'},
-                  {python: '3.11', dependencies: 'minimal'},
-                  {python: '3.8', dependencies: 'oldest'} ]
+        os: [ubuntu-latest]
+        python: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        include:
+          # Defaults to newest dependencies
+          - dependencies: 'newest'
+          # Oldest dependency tests
+          - python: '3.8'
+            dependencies: 'oldest'
+          - os: 'macos-latest'
+            python: '3.8'
+            dependencies: 'oldest'
+          - os: 'windows-latest'
+            python: '3.8'
+            dependencies: 'oldest'
+          # Newest version tests for non-Linux OS
+          - os: 'ubuntu-latest'
+            python: '3.12'
+            dependencies: 'newest'
+          - os: 'macos-latest'
+            python: '3.12'
+            dependencies: 'newest'
+          - os: 'windows-latest'
+            python: '3.12'
+            dependencies: 'newest'
+          # Minimal dependencies tests
+          - python: '3.12'
+            dependencies: 'minimal'
+          - os: 'macos-latest'
+            python: '3.12'
+            dependencies: 'minimal'
+          - os: 'windows-latest'
+            python: '3.12'
+            dependencies: 'minimal'
```

Comment:
If I'm reading this correctly (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixinclude) then we can introduce a key that will be overwritten to force the creation of a new entry. 
```suggestion
        os: [ubuntu-latest]
        python: ['3.8', '3.9', '3.10', '3.11', '3.12']
        default: 'true'
        include:
          # Defaults to newest dependencies
          - dependencies: 'newest'
          # Oldest dependency tests
          - python: '3.8'
            dependencies: 'oldest'
          - os: 'macos-latest'
            python: '3.8'
            dependencies: 'oldest'
          - os: 'windows-latest'
            python: '3.8'
            dependencies: 'oldest'
          # Newest version tests for non-Linux OS
          - os: 'ubuntu-latest'
            python: '3.12'
            dependencies: 'newest'
          - os: 'macos-latest'
            python: '3.12'
            dependencies: 'newest'
          - os: 'windows-latest'
            python: '3.12'
            dependencies: 'newest'
          # Minimal dependencies tests
          - default: 'false' 
            python: '3.12'
            dependencies: 'minimal'
          - os: 'macos-latest'
            python: '3.12'
            dependencies: 'minimal'
          - os: 'windows-latest'
            python: '3.12'
            dependencies: 'minimal'
```
If that doesn't work, you can review the docs and see if there is another way - or omit the matrix values entirely and list all tested configurations explicitly.